src/code/z_actor.c

```diff
@@ -1577,6 +1562,7 @@ u32 Actor_HasParent(Actor* actor, GlobalContext* globalCtx) {
     }
 }
 
+// Related to carrying an actor as well as giving the player an item?
 s32 func_8002F434(Actor* actor, GlobalContext* globalCtx, s32 getItemId, f32 xzRange, f32 yRange) {
```

Comment:
I think this function is the one used to give the player a GI item.
What do you think of some simple name like `Actor_GetItem`?

Reply:
I think there is already something like that, but no its not just that, this is also for carrying an item, i think the crate uses it or something really weird like that

Reply:
Doing a quick search, it seems only 3 actors explicitly call this function using `GI_NONE` to be hold by player (ishi, niw and tsubo). Every other hold-able actor calls `func_8002F580` to do that (which internally call this function with `GI_NONE`).

Considering that, I propose the following:
- Rename `func_8002F434` to `Actor_GetItem` and add an appropriate comment about the effects of passing `GI_NONE` and `GI_MAX`.
- Rename `func_8002F554` to `Actor_GetItemNearby`
- Rename `func_8002F580` to `Actor_HoldActor`

Reply:
What does passing in `GI_MAX` do?

Reply:
It is used to catch the actor in a bottle. It is used by insect, ice_hono, fish and elf.

Reply:
something like `Actor_Carry` makes more sense imo

Reply:
Yeah, that's sounds good too, I'm not very good naming stuff :sweat_smile:

Reply:
I was saying Actor_Carry for Actor_GetItem to clarify though. Link technically "carries" the item above his head

Reply:
> Carry
>  to take or support from one place to another; convey; transport
https://www.dictionary.com/browse/carry

Link isn't taking an item from one place to another above his head, so `Actor_Carry` would be a bit of a stretch.

Reply:
I feel like the reason that this function is so weird though has to be some idea like this though, like the devs somehow thinking that all this functionality is related 🤔

Reply:
Note:
Should he renamed to `Actor_PickUp` along with a comment explaining it's weird usage